### DISCUSSION TITLE
How to include name in cross references?

### DISCUSSION THREAD

**Discussion by drscotthawley** [quarto-dev/quarto-cli#9778] [Q&A] [answered]
If I provide a reference to a section, I want the name of that section to appear in the cross reference, e.g. with code like
```
# Further Study {#sec-further}

....
See the @sec-further section for more information
```

I want it  it so say, "See the _Further Study_ section for more information", not "See the _Section 9_  section for more information" How do we do that?  I'm reading the [documentation on Cross-References](https://quarto.org/docs/authoring/cross-references.html) but can't find any name-based  behavior described. 

I want it to grab whatever the section name is, so that even if I change the section name (e.g. to "Further Studies"), I won't have to remember to go back and re-name it everywhere in the text where it's referenced.

**Comment by sicabi** [comment]
Hello, I am also very interested in this topic. I haven't found a native Quarto solution, but only a Pandoc-lua filter solution written by @shafayetShafee, available in [StackOverflow](https://stackoverflow.com/questions/75071145/quarto-cross-references-persistent-cross-references-when-text-changes).

**Reply by shafayetShafee** [reply]
FYI, I have converted that answer to an extension [`nameref`](https://github.com/shafayetShafee/nameref). But this extension has an important [limitation](https://github.com/shafayetShafee/nameref?tab=readme-ov-file#limitations).

**Reply by sicabi** [reply]
Works quite well, thank you.

**Comment by cderv** [comment, marked as answer]
FWIW this is tracked in 
* https://github.com/quarto-dev/quarto-cli/issues/1196
* https://github.com/quarto-dev/quarto-cli/issues/4532

**Reply by jameshowison** [reply]
Followed these links and found that 

```
format: 
  html:
    number-sections: false
```

Seemed to get me the text rather than the section number.

**Reply by slowkow** [reply]
I tried `number-sections: false` and it does not make any difference for me. I always get `Section 5`, regardless of whether the option is set to true or false.

**Reply by mcanouil** [reply]
@slowkow Please when stating something "does not work", add a small and reproducible example or at least context.

**Reply by slowkow** [reply]
@mcanouil 

quarto 1.7.31

When I use `number-sections: true` I get `Section 1`.

When I use `number-sections: false` I still get `Section 1`. I would have expected to get `Methods` instead of `Section 1`.


_quarto.yml
```
project:
  output-dir: output/

format:
  html:
    number-sections: false
```

paper.qmd
```
---
title: My title
date: 2025-05-19
license: "CC BY"
---

{{< include includes/results.qmd >}}

{{< include includes/methods.qmd >}}
```

includes/results.qmd
```
# Results

## First section of results

See the methods here (@sec-methods).
```

includes/methods.qmd
```
# Methods {#sec-methods}

## First section of methods

Here are my methods.
```

![My title - Brave 2025-05-19_16-16-15](https://github.com/user-attachments/assets/ebb6db92-b730-4073-baef-257950197834)

**Reply by mcanouil** [reply]
Your example is really not a simple one.

```qmd
---
format:
  html:
    number-sections: false
---

# Methods {#sec-methods}

See the methods here (@sec-methods).
```

@slowkow Do note that the reply marked as the answer contained two issues that track this request.
The two issues are still opened.
Finally, if you follow the links, you can see that https://github.com/quarto-dev/quarto-cli/discussions/9778#discussioncomment-10730946 comments is only valid for Books projects.

**Reply by slowkow** [reply]
@mcanouil I tried your example, and it also produces `Section 1` instead of `Methods`.

**Reply by mcanouil** [reply]
Please fully read the replies.
